packages/share_plus/share_plus/android/src/main/kotlin/dev/fluttercommunity/plus/share/MethodCallHandler.kt

```diff
@@ -5,71 +5,74 @@ import io.flutter.plugin.common.MethodChannel
 import java.io.IOException
 
 /** Handles the method calls for the plugin.  */
-internal class MethodCallHandler(private val share: Share, private val manager: ShareSuccessManager) : MethodChannel.MethodCallHandler {
+internal class MethodCallHandler(
+    private val share: Share,
+    private val manager: ShareSuccessManager
+) : MethodChannel.MethodCallHandler {
 
-  override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
-    when (call.method) {
-      "share" -> {
-        expectMapArguments(call)
-        // Android does not support showing the share sheet at a particular point on screen.
-        share.share(
-          call.argument<Any>("text") as String,
-          call.argument<Any>("subject") as String?,
-          false,
-        )
-        result.success(null)
-      }
-      "shareFiles" -> {
-        expectMapArguments(call)
+    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
+        when (call.method) {
+            "share" -> {
+                expectMapArguments(call)
+                // Android does not support showing the share sheet at a particular point on screen.
+                share.share(
+                    call.argument<Any>("text") as String,
+                    call.argument<Any>("subject") as String?,
+                    false,
+                )
+                result.success(null)
+            }
+            "shareFiles" -> {
+                expectMapArguments(call)
 
-        // Android does not support showing the share sheet at a particular point on screen.
-        try {
-          share.shareFiles(
-            call.argument<List<String>>("paths")!!,
-            call.argument<List<String>?>("mimeTypes"),
-            call.argument<String?>("text"),
-            call.argument<String?>("subject"),
-            false,
-          )
-          result.success(null)
-        } catch (e: IOException) {
-          result.error(e.message, null, null)
-        }
-      }
-      "shareWithResult" -> {
-        expectMapArguments(call)
-        if(!manager.setCallback(result)) return
+                // Android does not support showing the share sheet at a particular point on screen.
+                try {
+                    share.shareFiles(
+                        call.argument<List<String>>("paths")!!,
+                        call.argument<List<String>?>("mimeTypes"),
+                        call.argument<String?>("text"),
+                        call.argument<String?>("subject"),
+                        false,
+                    )
+                    result.success(null)
+                } catch (e: IOException) {
+                    result.error("Share failed", e.message, null)
+                }
+            }
+            "shareWithResult" -> {
+                expectMapArguments(call)
```

Comment:
could these `*withResults(..)` methods extended with check like

```
if(Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP_MR1)
  result.notImplemented()
else ...
```

Then it won't be necessary to bump the `minSdkVersion` version for Android platform and it could be kept at the level supported in `3.x` version of plugin.

Reply:
I think so. If anyone can submit a separated PR with a that change, that would be awesome!